packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -417,6 +438,14 @@ export class Root extends Construct {
   }
 }
 
+/**
+ * Aspects that can be applied to Constructs.
+ */
+export interface IAspect {
+  readonly type: string;
```

Comment:
This type was included so we could filter on aspect types. For example, TagAspects check to see if there is another TagAspect that takes precedence. It's hard because parents and children can apply tags out of order and I had to rely on more than array order. There are other designs that could be done to make this work. For example, we could make the TagAspect collect all others TagAspects into an array and write a comparison function to always sort priority.